common/test/java/com/couchbase/lite/LiveQueryTest.java

```diff
@@ -51,108 +54,107 @@ public void testCreateBasicListener() throws InterruptedException {
+                latch1[n].countDown();
+            });
+
+        // listener 1 gets notified after observer subscribed
+        assertTrue(latch1[0].await(TOLERABLE_QUERY_RUN_TIME, TimeUnit.MILLISECONDS));
+
+        ListenerToken token2 = query.addChangeListener(
+            testSerialExecutor,
+            change -> {
+                int n = count[1]++;
+                latch2[n].countDown();
+            });
+
         try {
-            assertTrue(latch2[0].await(TOLERABLE_DELAY_MS, TimeUnit.MILLISECONDS));
+            // listener should get notify after query run time
+            assertTrue(latch2[0].await(TOLERABLE_QUERY_RUN_TIME, TimeUnit.MILLISECONDS));
 
-            // !!! THIS IS NOT TESTING WHAT IT CLAIMS TO TEST
-            // creation of token2 should not trigger first listener callback
-            assertFalse(latch1.await(2 * TOLERABLE_DELAY_MS, TimeUnit.MILLISECONDS));
+            // creation of the second listener should not trigger first listener callback
+            assertFalse(latch1[1].await(TOLERABLE_QUERY_RUN_TIME * 2, TimeUnit.MILLISECONDS));
 
             createDocNumbered(11);
 
-            // introducing change in database should trigger both listener callbacks
-            assertTrue(latch1.await(TOLERABLE_DELAY_MS, TimeUnit.MILLISECONDS));
-            assertTrue(latch2[1].await(TOLERABLE_DELAY_MS, TimeUnit.MILLISECONDS));
+            // introducing change in database should trigger both listener callbacks after the time it takes
+            // for core to wait and rerun query
+            assertTrue(latch1[1].await(TOLERABLE_DATABASE_CHANGE_DELAY, TimeUnit.MILLISECONDS));
+            assertTrue(latch2[1].await(TOLERABLE_DATABASE_CHANGE_DELAY, TimeUnit.MILLISECONDS));
         }
         finally {
             query.removeChangeListener(token1);
             query.removeChangeListener(token2);
         }
     }
 
-    // Creating a document that a query can see should cause an update
-    @Ignore("Test is flaky with 220ms delay")
-    @Test
-    public void testAddChangeToBasicLiveQuery() throws CouchbaseLiteException, InterruptedException {
-        final Query query = QueryBuilder
-            .select(SelectResult.expression(Meta.id))
-            .from(DataSource.database(baseTestDb))
-            .where(Expression.property(KEY).greaterThanOrEqualTo(Expression.intValue(0)))
-            .orderBy(Ordering.property(KEY).ascending());
-
-        final CountDownLatch latch = new CountDownLatch(1);
-
-        ListenerToken token = query.addChangeListener(testSerialExecutor, change -> latch.countDown());
-
-        createDocNumbered(10);
-
-        try { assertTrue(latch.await(TOLERABLE_DELAY_MS, TimeUnit.MILLISECONDS)); }
-        finally { query.removeChangeListener(token); }
-    }
-
-    // !!! What the ever-luvvin heck is this trying to test???
-    @Ignore("Test fail with 220ms delay")
+    // When a result set is closed, we should still be able to introduce a change
     @Test
```

Comment:
The name is pretty descriptive.  That's great.  I still think a little comment that described what the test is doing, would be excellent.